### PR TITLE
Add a simple identifier Scope class and a test.

### PR DESCRIPTION
**Pull request by matanlurey**

### CODE REVIEW

File: lib/src/scope.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
+part of code_builder;
+
+/// Determines an [Identifier] given where it appears and where it is from.
```
Comment (yjbanov) [inline lib/src/scope.dart]: Did you mean "depending on where it appears..."?

File: lib/src/scope.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  /// Create a default scope context.
+  ///
+  /// Actual implementation is _not_ guaranteed, only that all import prefixes
+  /// will be unique in a given scope (actual implementation may be naive).
+  factory Scope() = _IncrementingScope;
+
+  /// Create a context that does _not_ apply any scoping.
+  factory Scope.identity() = _IdentityScope;
+
+  /// Given a [symbol] and it's known [importUri], return an [Identifier].
```
Comment (yjbanov) [inline lib/src/scope.dart]: nit: s/it's/its/

File: lib/src/scope.dart (hunk 1 above)
Comment (matanlurey) [inline lib/src/scope.dart]: Yup! Done.

File: lib/src/scope.dart (hunk 2 above)
Comment (matanlurey) [inline lib/src/scope.dart]: 👍

**Pull request merged**